lib/tests/streamlit/state/widgets_test.py

```diff
@@ -70,6 +70,20 @@ def test_get_prev_widget_value(self):
         # not find anything.
         self.assertIsNone(widget_mgr.get_widget_value("trigger"))
 
+    def test_get_keyed_widget_values(self):
+        states = WidgetStates()
+        _create_widget("trigger", states).trigger_value = True
+        _create_widget("trigger2", states).trigger_value = True
+
+        deserializer = lambda x: x
+        widget_mgr = WidgetManager()
+        widget_mgr.set_widget_states(states)
+
+        widget_mgr.set_widget_attrs("trigger", True, None, deserializer, None, None)
```

Comment:
This kind of thing would be much cleaner if we passed values in by keyword and left most of them as None defaults. Or if the function's args aren't optional, create a factory for testing that fills in them with defaults.

Reply:
Done. The method initially wasn't initially written with optional args, but I don't believe there's any downside to doing so. Changed things to do this and cleaned up the test cases